packages/adminrouter/extra/src/README.md

```diff
@@ -535,8 +550,232 @@ DC/OS IAM relies on JSON Web Tokens for transmitting security information betwee
 components. Open repository flavour uses HS256 tokens, while EE relies on RS256.
 Test harness needs to have a way to create them for use in the tests. This is
 done by `test-harness/modules/mocker/jwt.py` module, which together with
-`repo_is_ee` fixture provides abstracts away the type of the
-token used by Admin Router itself.
+`repo_is_ee` fixture provides abstracts away the type of the token used by
+Admin Router itself.
+
+### Tests code reuse and deduplication
+Lots of endpoints exposed by Admin Router share the common behaviour. Usually, it
+is something simple like testing for correct headers, right upstream, etc...
+This results in tests covering them being very repetitive.
+
+In order to keep the code DRY, common code was extracted into a library with
+generic tests & helpers and a testing framework using YAML configuration files
+was developed.
+
+#### Generic tests
+Tests shared by most of the endpoints are as follow:
```

Comment:
It is true that this may get outdated, but IMO it is worth it - I mean it as a starting point/bootstrap for people trying to write tests using test harness.